### PR TITLE
[prettier] Ban array recursion in doc.builder.Doc array

### PR DESCRIPTION
**Pull request by ghost**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/prettier/prettier/blob/main/commands.md#prettiers-intermediate-representation-doc
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

> arrays are used to concatenate a list of docs to be printed sequentially into a single doc

Refs: #54526 (unrelated PR that needed this fix)

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## prettier ([<kbd>unpkg</kbd>](https://unpkg.com/browse/prettier@latest/))
was missing the following properties:
1. diffChars
1. characterDiff
1. diffWords
1. diffWordsWithSpace
1. wordDiff

<details>
<summary>as well as these 263 other properties...</summary>
<p>diffLines, diffTrimmedLines, lineDiff, diffSentences, sentenceDiff, diffCss, cssDiff, diffJson, canonicalize, jsonDiff, diffArrays, arrayDiff, Diff, applyPatch, applyPatches, parsePatch, merge, structuredPatch, createTwoFilesPatch, createPatch, convertChangesToDMP, convertChangesToXML, re, src, t, tildeTrimReplace, caretTrimReplace, comparatorTrimReplace, outdent, apiDescriptor, supportsColor, isIdentifierName, isIdentifierChar, isIdentifierStart, isReservedWord, isStrictBindOnlyReservedWord, isStrictBindReservedWord, isStrictReservedWord, isKeyword, posix, win32, SEMVER_SPEC_VERSION, parse, valid, clean, SemVer, inc, diff, compareIdentifiers, rcompareIdentifiers, major, minor, patch, compare, compareLoose, rcompare, sort, rsort, gt, lt, eq, neq, gte, lte, cmp, Comparator, Range, toComparators, satisfies, maxSatisfying, minSatisfying, minVersion, validRange, ltr, gtr, outside, prerelease, intersects, coerce, minimatch, flatten, splitWhen, isEnoentCodeError, createDirentFromStats, unixify, makeAbsolute, escape, removeLeadingDotSegment, isInteger, find, exceedsLimit, escapeNode, encloseBrace, isInvalidBrace, isOpenOrClose, reduce, isObject, hasRegexChars, isRegexChar, escapeRegex, toPosixSlashes, removeBackslashes, supportsLookbehinds, isWindows, escapeLast, removePrefix, wrapOutput, isStaticPattern, isDynamicPattern, convertToPositivePattern, convertToNegativePattern, isNegativePattern, isPositivePattern, getNegativePatterns, getPositivePatterns, getPatternsInsideCurrentDirectory, getPatternsOutsideCurrentDirectory, isPatternRelatedToParentDirectory, getBaseDirectory, hasGlobStar, endsWithSlashGlobStar, isAffectDepthOfReadingPattern, expandPatternsWithBraceExpansion, expandBraceExpansion, getPatternParts, makeRe, convertPatternsToRe, matchAny, isString, isEmpty, array, errno, fs, path, pattern, stream, string, generate, convertPatternsToTasks, getNegativePatternsAsPositive, groupPatternsByBaseDirectory, convertPatternGroupsToTasks, convertPatternGroupToTask, read, FILE_SYSTEM_ADAPTER, createFileSystemAdapter, Settings, stat, statSync, IS_SUPPORT_READDIR_WITH_FILE_TYPES, joinPathSegments, readdirWithFileTypes, readdir, scandir, scandirSync, isFatalError, isAppliedFilter, replacePathSegmentSeparator, walk, walkSync, walkStream, DEFAULT_FILE_SYSTEM_ADAPTER, ast, code, keyword, $EOF, $BSPACE, $TAB, $LF, $VTAB, $FF, $CR, $SPACE, $BANG, $DQ, $HASH, $$, $PERCENT, $AMPERSAND, $SQ, $LPAREN, $RPAREN, $STAR, $PLUS, $COMMA, $MINUS, $PERIOD, $SLASH, $COLON, $SEMICOLON, $LT, $EQ, $GT, $QUESTION, $0, $7, $9, $A, $E, $F, $X, $Z, $LBRACKET, $BACKSLASH, $RBRACKET, $CARET, $_, $a, $b, $e, $f, $n, $r, $t, $u, $v, $x, $z, $LBRACE, $BAR, $RBRACE, $NBSP, $PIPE, $TILDA, $AT, $BT, isWhitespace, isDigit, isAsciiLetter, isAsciiHexDigit, isNewLine, isOctalDigit, sanitizeIdentifier, identifierName, identifierModuleUrl, viewClassName, rendererTypeName, hostViewClassName, componentFactoryName, CompileSummaryKind, tokenName, tokenReference, CompileStylesheetMetadata, CompileTemplateMetadata, CompileDirectiveMetadata, CompilePipeMetadata, CompileShallowModuleMetadata, CompileNgModuleMetadata, TransitiveCompileNgModuleMetadata, ProviderMeta, templateSourceUrl, sharedStylesheetJitUrl, ngModuleJitUrl, templateJitUrl, ParseLocation, ParseSourceFile, ParseSourceSpan, EMPTY_PARSE_LOCATION, EMPTY_SOURCE_SPAN, ParseErrorLevel, ParseError, typeSourceSpan, r3JitTypeSourceSpan</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 2970f96f931f1c83ba550d538f581d86a320f604
</p>

File: types/prettier/index.d.ts [hunk 1]
```diff
@@ -569,7 +569,7 @@ export namespace doc {
             | LineSuffix
             | LineSuffixBoundary
             | Trim;
-        type Doc = string | Doc[] | DocCommand;
+        type Doc = string | DocCommand | Array<string | DocCommand>;
```
Comment (JounQin) [inline types/prettier/index.d.ts:572]: I don't understand why you'd like to change this, the original type definition is just as the same as https://github.com/prettier/prettier/blob/main/commands.md#prettiers-intermediate-representation-doc

**Review by JounQin — CHANGES_REQUESTED** [review]

File: types/prettier/index.d.ts (hunk 1 above)
Comment (SNDST00M) [reply types/prettier/index.d.ts:572]: Is that intended to allow recursion - array of array etc?

Comment (thorn0) [reply types/prettier/index.d.ts:572]: yes

**Pull request closed without merging**